util/CosmosHelper.js

```diff
@@ -122,66 +134,84 @@ export async function getTransferInfo(txHash, opt) {
 }
 
 export async function queryLikeCoinBalance(addr) {
-  if (!api) await initCosmos();
-  const account = await api.get.account(addr);
-  const [amount] = account.coins.filter(coin => coin.denom === COSMOS_DENOM);
+  if (!cosmosClient) await initCosmosClient();
+  const account = await cosmosClient.getAccount(addr);
+  const [amount] = account.balance.filter(balance => balance.denom === COSMOS_DENOM);
   if (!amount) return 0;
   return amountToLIKE(amount);
 }
 
-export async function sendTx(msgCallPromise, signer, {
-  gasPrices = DEFAULT_GAS_PRICE,
-  memo,
-  simulate: isSimulate,
-} = {}) {
-  const { simulate, send } = await msgCallPromise;
-  const gas = (await simulate({ memo })).toString();
-  if (isSimulate) return { gas, gasPrices };
-  const { hash, included } = await send({ gas, gasPrices, memo }, signer);
-  return {
-    txHash: hash,
-    included,
-    gas,
-    gasPrices,
-  };
-}
-
 export async function transfer({
   from,
   to,
   value,
   memo,
-}, signer, { simulate = false } = {}) {
-  if (!api) await initCosmos();
+}, signer) {
   const amount = LIKEToAmount(value);
-  const msgPromise = api.MsgSend(from, { toAddress: to, amounts: [amount] });
-  return sendTx(msgPromise, signer, { memo, simulate });
+  const fee = {
+    amount: [{
+      denom: 'nanolike',
+      amount: '1000',
+    }],
+    gas: '44000',
+  };
+  const defaultGasPrice = GasPrice.fromString('1000nanolike');
+  if (!signingCosmosClient) {
+    await initSigningCosmosClient(COSMOS_RESTFUL_API, from, signer, defaultGasPrice, {}, 'block');
+  }
+  const sendMsg = {
+    type: 'cosmos-sdk/MsgSend',
+    value: {
+      from_address: from,
+      to_address: to,
+      amount: [amount],
+    },
+  };
+  const broadcastedTx = await signingCosmosClient.signAndBroadcast([sendMsg], fee, memo);
```

Comment:
Does this break if user select another gas price  in keplr UI,other than the hardcoded `fee` here?

Reply:
No, tested both flow: authcore & Keplr. with send(==1) & sendMultiple(>1 receiver). Gas price will increase based on the chosen gas price by user.